src/planner/operator/logical_insert.cpp

```diff
@@ -3,12 +3,48 @@
 namespace duckdb {
 
 void LogicalInsert::Serialize(FieldWriter &writer) const {
-	throw NotImplementedException(LogicalOperatorToString(type));
+	writer.WriteField<idx_t>(insert_values.size());
+	for (auto &entry : insert_values) {
+		writer.WriteSerializableList(entry);
+	}
+	writer.WriteList<idx_t>(column_index_map);
+	writer.WriteRegularSerializableList(expected_types);
+	table->Serialize(writer.GetSerializer());
+	writer.WriteField(table_index);
+	writer.WriteField(return_chunk);
+	writer.WriteSerializableList(bound_defaults);
+
 }
 
 unique_ptr<LogicalOperator> LogicalInsert::Deserialize(ClientContext &context, LogicalOperatorType type,
                                                        FieldReader &reader) {
-	throw NotImplementedException(LogicalOperatorToString(type));
+	auto insert_values_size = reader.ReadRequired<idx_t>();
+	vector<vector<unique_ptr<Expression>>> insert_values;
+	for (idx_t i = 0; i < insert_values_size; i++) {
```

Comment:
Super picky, but I usually prefer postfix operator here:
```suggestion
	for (idx_t i = 0; i < insert_values_size; ++i) {
```
@Mytherin do you have some rules for these kind of coding style too?

Reply:
I usually do `i++` but we don't enforce a style there